@xen-orchestra/lite/src/App/Infrastructure.tsx

```diff
@@ -44,21 +49,31 @@ interface Effects {
 
 interface Computed {}
 
+const selectedNodestoArray = (nodes: Array<string> | string | undefined) =>
+  nodes === undefined ? undefined : Array.isArray(nodes) ? nodes : [nodes]
+
 const Infrastructure = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
   {
     initialState: props => ({
       selectedVm: props.location.pathname.split('/')[3],
     }),
+    computed: {
+      selectedPool: state => state.objectsByType?.get('pool')?.keySeq().first(),
+    },
   },
-  ({ state: { selectedVm } }) => (
+  ({ state: { selectedVm, selectedPool }, ...props }) => (
     <Container>
       <LeftPanel>
-        <TreeView defaultSelectedNodes={selectedVm === undefined ? undefined : [selectedVm]} />
+        <TreeView
+          defaultSelectedNodes={selectedNodestoArray(
+            props.location.pathname.startsWith('/infrastructure/pools') ? selectedPool : selectedVm
+          )}
+        />
       </LeftPanel>
       <MainPanel>
         <Switch>
-          <Route exact path='/infrastructure'>
-            Select a VM
+          <Route exact path={`/infrastructure/pools/${selectedPool}/dashboard`}>
```

Comment:
Should it be `pool` or `pools`, since there's only one pool? Did we discuss it somewhere, already? I don't remember.